hid-uclogic.c

```diff
@@ -24,6 +24,15 @@
 #include "compat.h"
 #include <linux/version.h>
 
+static inline s32 le24_to_cpu(const __u8* ptr)
+{
+	return ptr[0] | (ptr[1] << 8UL) | (ptr[2] << 16UL);
+}
+
+#define UCLOGIC_RDESC_ID (0x64)
+#define UCLOGIC_HIRES_RDESC_ID (0xC8)
+#define UCLOGIC_HIRES_RDESC_LENGTH (18)
```

Comment:
Naming these "RDESC" is confusing. They are not "report descriptors", they are "string descriptors", specifically "string descriptors containing device parameters which are also configuring the device". I would shorten that to something like "PRM_STR".